crates/cheatnet/src/state.rs

```diff
@@ -121,10 +140,21 @@ impl StateReader for ExtendedStateReader {
 }
 
 pub enum CheatStatus<T> {
-    Cheated(T),
+    Cheated(T, CheatSpan),
     Uncheated,
 }
 
+impl<T> CheatStatus<T> {
+    pub fn decrement_cheat_span(&mut self) {
+        if let CheatStatus::Cheated(_, CheatSpan::Number(n)) = self {
```

Comment:
This should probably panic otherwise, or return result

Reply:
That's actually the logic on this - `CheatStatus` becomes `Uncheated`, once its cheat span goes down to 0.